daemon/api.go

```diff
@@ -2549,8 +2550,12 @@ func postDebug(c *Command, r *http.Request, user *auth.UserState) Response {
 			}
 		}
 		sort.Strings(status.Unreachable)
-
 		return SyncResponse(status, nil)
+	case "refresh-catalogs":
+		if err := snapstateRefreshCatalogs(st); err != nil {
+			return InternalError("cannot refresh catalogs: %s", err)
```

Comment:
you could actually return a much richer error (there are a bunch of `switch err {` in `api.go` for inspiration). It might not be worth it for a debug command though...